src/Shared/CertificateGeneration/UnixCertificateManager.cs

```diff
@@ -44,29 +162,750 @@ protected override X509Certificate2 SaveCertificateCore(X509Certificate2 certifi
         return certificate;
     }
 
-    internal override CheckCertificateStateResult CheckCertificateState(X509Certificate2 candidate, bool interactive)
+    internal override CheckCertificateStateResult CheckCertificateState(X509Certificate2 candidate)
     {
         // Return true as we don't perform any check.
+        // This is about checking storage, not trust.
         return new CheckCertificateStateResult(true, null);
     }
 
     internal override void CorrectCertificateState(X509Certificate2 candidate)
     {
         // Do nothing since we don't have anything to check here.
+        // This is about correcting storage, not trust.
     }
 
     protected override bool IsExportable(X509Certificate2 c) => true;
 
-    protected override void TrustCertificateCore(X509Certificate2 certificate) =>
-        throw new InvalidOperationException("Trusting the certificate is not supported on linux");
+    protected override TrustLevel TrustCertificateCore(X509Certificate2 certificate)
+    {
+        var sawTrustFailure = false;
+        var sawTrustSuccess = false;
+
+        using var store = new X509Store(StoreName.Root, StoreLocation.CurrentUser);
+        store.Open(OpenFlags.ReadWrite);
+
+        if (TryFindCertificateInStore(store, certificate, out _))
+        {
+            sawTrustSuccess = true;
+        }
+        else
+        {
+            try
+            {
+                using var publicCertificate = X509CertificateLoader.LoadCertificate(certificate.Export(X509ContentType.Cert));
+                // FriendlyName is Windows-only, so we don't set it here.
+                store.Add(publicCertificate);
+                Log.UnixDotnetTrustSucceeded();
+                sawTrustSuccess = true;
+            }
+            catch (Exception ex)
+            {
+                sawTrustFailure = true;
+                Log.UnixDotnetTrustException(ex.Message);
+            }
+        }
+
+        var homeDirectory = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+
+        // Rather than create a temporary file we'll have to clean up, we prefer to export the dev cert
+        // to its final location in the OpenSSL directory.  As a result, any failure up until that point
+        // is fatal (i.e. we can't trust the cert in other locations).
+
+        var certDir = GetOpenSslCertificateDirectory(homeDirectory)!; // May not exist
```

Comment:
Make sure this directory is not world readable.

Reply:
It's probably easier to rebase https://github.com/dotnet/aspnetcore/pull/56985 on top of this than vice versa.

Reply:
Done.